src/main/java/gregtech/api/capability/impl/FuelRecipeLogic.java

```diff
@@ -111,18 +110,32 @@ public int getNetworkID() {
     @Override
     public void update() {
         if (getMetaTileEntity().getWorld().isRemote) return;
+
         if (workingEnabled) {
-            if (recipeDurationLeft > 0) {
-                if (energyContainer.get().getEnergyCanBeInserted() >=
-                    recipeOutputVoltage || shouldVoidExcessiveEnergy()) {
-                    energyContainer.get().addEnergy(recipeOutputVoltage);
-                    if (--this.recipeDurationLeft == 0) {
-                        this.wasActiveAndNeedsUpdate = true;
-                    }
-                }
+
+            /* we need to compute those two before doing anything
+             * to avoid loosing 1 tick of production
+             */
+
+            boolean canProduce = canProduceEnergy() &&
+                    (energyContainer.get().getEnergyCanBeInserted() >= calculateRecipeOutputVoltage() ||
+                            shouldVoidExcessiveEnergy());
+
+            boolean canConsume = canConsumeFuel() && (shouldConsumeWhenNotProducing() || canProduce);
+
+            if (canConsume) {
+                --this.recipeDurationLeft;
             }
-            if (recipeDurationLeft == 0 && isReadyForRecipes()) {
-                tryAcquireNewRecipe();
+
+            if (canProduce) {
+                energyContainer.get().addEnergy(calculateRecipeOutputVoltage());
```

Comment:
You should probably call calculateRecipeOutputVoltage every tick and write result into the backing field, then use it's value here and in the getOutputVoltage method.

Reply:
why would I ever do that ? I mean yes that would work but how would you me to name that ? and why would a getter called getXXX returning something else than XXX but YY a value created from another methods ? it doesn't makes sense

Reply:
Because ideally you want value that was ACTUALLY used to generate energy this tick exposed via getRecipeOutputVoltage, and not recomputed every time getter is called, potentially producing unwanted side effects and giving incorrect results when other fields are changed.

Reply:
no need for caps lock.

yes that is a  solution and a problem. please when you are proposing changes which fix a problem that haven't been pointed out yet point it out first then propose the solution this way makes more sense.